### DISCUSSION TITLE
GeoNode 5 Roadmap (and beyond)

### DISCUSSION THREAD

**Discussion by giohappy** [GeoNode/geonode#12796] [Announcements]
2025 will bring a new major version for GeoNode, an opportunity to refactor and bring new core features to this great platform, which is growing fast thanks to its community of contributors and customers.

We want to share the plans and ideas we're discussing along the road. 
Of course, the list reported below is only a fraction of the many things we would like to do but, hey, step by step...

A list of activities are already on fire:

 - :fire: A brand new **Metadata service** (https://github.com/GeoNode/geonode/issues/11511), based on JSON Schema, which includes a new highly customizable metadata model, a new metadata API, and a new editor that autoconfigured based on the API.
 - :fire: The upgrade to PyCSW3
 - :fire: Remove all the current legacy code and templates for metadata editing. The settings currently available inside the Metadata page are being moved to a new "Management" tab of a resource sidebar (you may have already spotted it in the development demo instance). The Attributes editor will use [related functionality](https://docs.mapstore.geosolutionsgroup.com/en/latest/user-guide/layer-settings/#feature-info-form) provided by the MapStore client 
 - :fire: A new home page and client layout, that brings more options for customizing the catalog, and sub-catalogs. The re-design is done in collaboration with the MapStore team. You can take a look at what's cooking at https://github.com/geosolutions-it/MapStore2/issues/10631
 - :fire: A new Django Admin area, which will be renamed to "GeoNode Back office", based on [unfold](https://unfoldadmin.com/). The plan is to revamp the admin area and give it a central role also for non-admin users.
 - :fire: The upgrade to Geoserver 2.26.x
 - :alarm_clock: [The removal of the deprecated `monitoring` module. ](https://github.com/GeoNode/geonode/issues/13110)
 - :alarm_clock: Removal of legacy project scripts and configurations (distribution packaging, playbooks, pavement, etc.)
 - :alarm_clock: Removal of `setup.cfg` and `requirements.txt` duplications, and review of pinned dependencies

On the side of deployment and security, we are ready to launch a new infrastructure for the demo instances, based on Kubernetes and GitHub deployment automation. This will allow us to experiment more with k8s deployments for GeoNode, its Helm charts and, last but not least, rootless deployments.

We also plan to work on the side of performances. Once most of the legacy code is dropped, we will have the opportunity to refactor and optimize areas that are currently tied to legacy architectures. These include optimizing the code and queries behind the principal REST API endpoints (resources and permissions), and the introduction of a new caching mechanism. 

Many more things are under discussion, sorted by priority and feasibility (not importance!):

 - https://github.com/GeoNode/geonode/issues/13108
 - Removal of all legacy Django templates and views for the client. One by one every frontend functionality that depends on legacy templates will either be moved to the client or the (new) Back office. This includes:
   - User profile (including notifications and user activity)
   - Groups management
   - Announcements (which are currently broken)
 - For the reasons above, and lack of maintenance, **the following functionalities are planned to be removed during the course of GeoNode 5 if nobody raises his hands against it**: https://github.com/GeoNode/geonode/issues/13111
   -  Messages
   -  Popular count
-  We would like to resurrect Activity streams since they provide valuable insights on GeoNode usage and auditing. 
- We hope we will have the opportunity to work on implementing analytics support inside the client. Tests done with bindings of the MapStore API to analytics clients for Matomo, Google Analytics, and Plausible look promising

These are not planned yet. We're open to contributions, ideas, and... sponsoring! :blush:

Thanks to all the people and companies contributing to the project. We hope to extend the participation even more and this discussion goes in that direction.

**Comment by t-book** [comment]
Wow this is big thanks @giohappy !

**Comment by ridoo** [comment]
@giohappy Do you have plans to create a pre-release once you approach the finishing line of v5? Just let us know.

**Comment by giohappy** [comment]
Hi @ridoo.
Yes, definitely. It's only a matter of finding the time :smile:

The plan is to create branch 5.0.x very soon to start stabilizing GeoNode 5.
It's a bit too early for the feature freeze because soon the migration to Redis and the upgrade to Django 5 will be merged. 
The goal is to start a feature freeze in the second half of October and, hopefully, have a pre-release in the first half of November. 
If everything goes well, we would like to release GeoNode 5 by the end of November or the beginning of December. 
This is the happy path....
